### PR TITLE
[Merged by Bors] - feat(FieldTheory/AlgebraicClosure): A polynomial splits in `E` implies it splits in `algebraicClosure F E`

### PR DESCRIPTION
**Pull request by jjdishere**
Add a lemma saying a polynomial splits in `E` implies it splits in `algebraicClosure F E`.

Co-authored-by: Junyan Xu <junyanxu.math@gmail.com>

---

- [x] depends on: #16771

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by erdOne** [comment]
A 200+ line difference is definitely not "easy"

**Comment by jjdishere** [comment]
> A 200+ line difference is definitely not "easy"
> 
> Edit:ah it's dependent on some other PR. Maybe add easy after the previous is merged.

Ah sorry! This is one single lemma after #16771. I'll add easy after the previous is merged in the future.

File: Mathlib/FieldTheory/AlgebraicClosure.lean [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    (g : (minpoly F x).Splits (algebraMap F E)) :
+    (minpoly F x).Splits (algebraMap F (algebraicClosure F E)) := by
+    apply Polynomial.splits_of_comp (algebraMap F (algebraicClosure F E))
+      (algebraMap (algebraicClosure F E) E) g
+    simp only [mem_roots']
+    intro r ⟨_, hrm⟩
+    simp only [IsRoot.def, Polynomial.eval_map] at hrm
+    simp only [RingHom.mem_range, IntermediateField.algebraMap_apply, Subtype.exists, exists_prop',
+      nonempty_prop, exists_eq_right]
+    exact ⟨minpoly F x, ⟨minpoly.monic h, hrm⟩⟩
```
[the remark this answers is not in the document]
Comment (jjdishere) [reply Mathlib/FieldTheory/AlgebraicClosure.lean:214]: Thank you very much! I'll generalize it at once.

**Review by j-loreaux — APPROVED** [review]
LGTM. Thanks!

maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**